src/app/loadout-builder/generated-sets/CompareDrawer.tsx

```diff
@@ -67,10 +67,12 @@ function createLoadoutUsingLOItems(
       }
 
       for (const item of draftLoadout.items) {
-        const existingLoadoutItem = loadoutItems.find((i) => i.item.id === item.id);
+        // Accessing id is safe: Armor is always instanced
+        const existingLoadoutItem = loadoutArmor.find((i) => i.item.id === item.id);
         const hasBeenReplaced =
           (existingLoadoutItem &&
             setItems.some((i) => i.bucket.hash === existingLoadoutItem.item.bucket.hash)) ||
+          // TODO: This fails to overwrite a migrated or received subclass!
```

Comment:
This whole function is a bit painful and should probably use a different approach.